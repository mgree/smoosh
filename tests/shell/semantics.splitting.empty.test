arg_len() {
    echo "$#"
}

echo Default:
# Default IFS
IFS=' 	
'

arg_len $(echo) $(echo) $(echo) $(echo)
arg_len $(echo " ") $(echo " ") $(echo " ") $(echo " ")
arg_len "$(echo)" "$(echo)" "$(echo)" "$(echo)"
arg_len "$(echo " ")" "$(echo " ")" "$(echo " ")" "$(echo " ")"

echo Colon:
# Set IFS.
IFS=":"

arg_len $(echo) $(echo) $(echo) $(echo)
arg_len $(echo " ") $(echo " ") $(echo " ") $(echo " ")
arg_len "$(echo)" "$(echo)" "$(echo)" "$(echo)"
arg_len "$(echo " ")" "$(echo " ")" "$(echo " ")" "$(echo " ")"

echo Unset:
# Unset IFS is the same as the default.
unset IFS

arg_len $(echo) $(echo) $(echo) $(echo)
arg_len $(echo " ") $(echo " ") $(echo " ") $(echo " ")
arg_len "$(echo)" "$(echo)" "$(echo)" "$(echo)"
arg_len "$(echo " ")" "$(echo " ")" "$(echo " ")" "$(echo " ")"

$(echo) echo OK
