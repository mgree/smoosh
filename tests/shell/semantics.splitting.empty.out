Default:
0
0
4
4
Colon:
0
4
4
4
Unset:
0
0
4
4
OK
